File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
def solution(word):
    answer = 0
    words = ['A', 'E','I', 'O', 'U']
    tempArr = []
    def rec(depth, curWord):
        if depth <= 5:
            tempArr.append(curWord)
            for i in range(5):        
                rec(depth + 1, curWord + words[i])
    rec(0, '')
    # print(tempArr)
    return tempArr.index(word)
    # return answer

File: 프로그래머스/2/84512. 모음 사전/모음 사전.js



function solution(word) {
    let answer = 0;
    let words = ['A', 'E', 'I', 'O', 'U'];
    let tempArr = [];
    
    function DFS(cur, depth) {
        if (depth <= 5) {
            tempArr.push(cur) // 현재 문자열을 추가

            for (let i = 0; i < words.length; i ++) {
                DFS(cur + words[i], depth + 1)
            }
        }
    }
    DFS('', 0)
    console.log(tempArr)
    answer = tempArr.indexOf(word)
    return answer;
}